Sming/SmingCore/Network/Http/HttpRequest.h

```diff
@@ -56,7 +56,8 @@ class HttpRequest
 		return new HttpRequest(*this);
 	}
 
-	HttpRequest& operator=(const HttpRequest& rhs);
+	/** @deprecated Please use `clone()` instead */
+	HttpRequest& operator=(const HttpRequest& rhs) SMING_DEPRECATED;
```

Comment:
The copy operator provided is actually unsafe as it doesn't take copies of allocated member variables so I've marked it deprecated. I could go further and make it private, which would prevent applications building if it's used. Thoughts?

Reply:
>  I could go further and make it private, which would prevent applications building if it's used. 

For now deprecate it only.

> I think probably get rid of the warning in the code and document the copy constructor instead.

Sounds good.